internal/services/nginx/nginx_deployment_resource.go

```diff
@@ -262,6 +285,24 @@ func (m DeploymentResource) Arguments() map[string]*pluginsdk.Schema {
 				}, false),
 		},
 
+		"web_application_firewall_settings": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"activation_state": {
```

Comment:
if the block becomes `web_application_firewall` this would be better as `activated` or `enabled`? which would make it more consistent with the rest of the provided

Reply:
addtionally is this the only setting that will be here or will there be further items

Reply:
I am fine with either naming convention but would prefer what's already there as the WAF experience sounds fine to me. Will sync up with @arpith-f5 once to see if we want to change this.